src/Hamiltonians/HubbardMomSpace.jl

```diff
@@ -0,0 +1,639 @@
+end
+
+@inline function extended_mom_transfer_diag(map::ModeMap, g::CubicGrid{D,S}, ::Nothing, w) where {D,S}
+    if iszero(w)
+        return 0.0
+    end
+    onproduct = 0
+    for i in 1:length(map)
+        occ_i = map[i].occnum
+        onproduct += occ_i * (occ_i - 1) * (w*D)
+        for j in 1:i-1
+            occ_j = map[j].occnum
+            q = g[map[i].mode] - g[map[j].mode]
+            onproduct += 2 * occ_i * occ_j * (D + _cosin_sum(q, S))
+        end
+    end
+    return onproduct * w
+end
+
+@inline function extended_mom_transfer_diag(map::ModeMap, ::CubicGrid, u, ::Nothing)
+    if iszero(u)
+        return 0
+    end
+    onproduct = 0
+    for i in 1:length(map)
+        occ_i = map[i].occnum
+        onproduct += occ_i * (occ_i - 1)
+        for j in 1:i-1
+            occ_j = map[j].occnum
+            onproduct += 4 * occ_i * occ_j
+        end
+    end
+    return onproduct * u
+end
+
+@inline function extended_mom_transfer_diag(map::FermiOccupiedModeMap, g::CubicGrid{D,S}, _, w) where {D,S}
+    if iszero(w)
+        return 0
+    end
+    onproduct = 0
+    for i in 1:length(map)
+        occ_i = map[i].occnum
+        onproduct += occ_i * (occ_i - 1)
+        for j in 1:i-1
+            occ_j = map[j].occnum
+            q = g[map[i].mode] - g[map[j].mode]
+            onproduct += 2 * occ_i * occ_j * (D - _cosin_sum(q, S))
+        end
+    end
+    return onproduct*w
+end
+
+@inline extended_mom_transfer_diag(::FermiOccupiedModeMap, ::CubicGrid, _, ::Nothing) = 0
+
+@inline function extended_mom_transfer_diag(map1::ModeMap, map2::ModeMap) 
+    onproduct = 0
+    for i in map1
+        occ_i = i.occnum
+        for j in map2
+            occ_j = j.occnum
+            onproduct += occ_i * occ_j
+        end
+    end
+    return onproduct
+end
+
+@inline extended_mom_transfer_diag(map1::FermiOccupiedModeMap, map2::FermiOccupiedModeMap) = length(map1) * length(map2)
+
+function _cosin_sum(q::SVector{D}, S::NTuple{D}) where {D}
+    onproduct = 0.0
+    for i in 1:D
+        onproduct += cos(q[i] * 2π / S[i])
+    end
+    return onproduct
+end
+
+"""
+    _mom_interactions_dig(component, g)
+Calculate the interaction terms for a given component of a multi-component Fock state address in momentum space.
+`component` is a tuple of all combination between the pair of relevant components of the multi-component Fock state 
+for which the interaction term needs to be calculated,
+
+'''math
+    \\hat{H}_\\text{int} = \\frac{1}{2}\\sum_{p,q,σ,σ'} V_{σσ'} \\hat{b}^†_{pσ} \\hat{b}^†_{qσ'} \\hat{b}^†_{qσ'} \\hat{b}_{pσ}
+'''
+
+where `V_{σσ}' is the interaction coefficent that depends on  `u_{σσ'}' and `w_{σσ'}'. `g` is the geometry of the lattice.
+
+"""
+
+function _mom_interactions_dig(component::Tuple, g::CubicGrid{D,S}) where {D,S}
+    onproduct = 0.0
+    for data in component
+        u = data.u
+        w = data.w
+        if !(isnothing(u) && isnothing(w)) 
+            Index = component_index(data)
```

Comment:
```suggestion
            index = component_index(data)
```
CamelCase is only for types and type parameters